### DISCUSSION TITLE
Can't get federation to work

### DISCUSSION THREAD

**Discussion by robp94** [smallrye/smallrye-graphql#1709] [Q&A]
I tried the integrated federation with quarkus 2.16.0, but it wouldn't work for me. Maybe you guys @t1 @jmartisk have an idea?

The schema generation looks good to me.

```
union _Entity = BusinessPartner

type Query {
  _entities(representations: [_Any!]!): [_Entity]!
  _service: _Service!
  businessPartner(id: ID): BusinessPartner
...
}
type BusinessPartner @key(fields : "id") {
  id: ID!
...
}
```
But on using federation I get an exception.
```
  "exception": {
    "refId": 1,
    "exceptionType": "java.lang.RuntimeException",
    "message": "can\u0027t resolve federated entity type io.smallrye.context.Jdk9CompletableFutureWrapper",
    "frames": [
      {
        "class": "io.smallrye.graphql.bootstrap.Bootstrap",
        "method": "lambda$fetchEntityType$2",
        "line": 214
      },
```

I looked a bit into it, but currently I don't really understand the problem.

![Federation1](https://user-images.githubusercontent.com/42009775/215099944-9a1bdd60-2630-41d5-a498-04a89d30894d.png)

**Comment by t1** [comment]
Maybe you return a `Multi`... and the federation part doesn't support that, yet. We had that discussion already somewhere maybe two weeks ago.

**Reply by robp94** [reply]
No, there is no multi return.

**Comment by steffenthomasEMP** [comment]
Hey @t1 and @jmartisk, 
I'm facing issues with federation, as well. Maybe you can give me some hints...because as I followed docs at [smallrye federation](https://smallrye.io/smallrye-graphql/1.9.4/federation/), I'm not able to `extend` a type from another. 
I use:
* quarkus:3.0.0.Beta1
* smallrye-graphql-schema-model:2.1.2

I have following:
-> item-resolver
```
type Item {
  id: ID!
  name: String!
}
```

-> my-resolver
```
type Item {
  id: ID!
  name: String!
  pathName: String!
}
```

but implementation is:
```
@NoArgsConstructor@Data // lombok stuff...
@Extends @Key(fields = "id")
public class Item {
    
    @Id
    @NonNull
    private String id;
    
    @NonNull
    private String name;

    @NonNull
    @AdaptWith(MyUrlAdapter.class)
    private MyUrl pathName;

}
```

I expect from the docs that I should have a schema like:
```
type Item @extends @key(fields : "id"){
  id: ID!
  name: String!
  pathName: String!
}
```

If you need any further information about any external libs or something else...let me know. 
I've also activated federation in application.yml like:
```
quarkus:
    smallrye-graphql:
    federation:
      enabled: true
```
And checked that by debugging into SmallRyeGraphQLProcessor#activateFederation...the config is corretly enabled...so, I'm stuck right now, because the federated supergraph validation is failing because of missing schema directive 'extends' (I guess)

Best
Steffen

**Reply by steffenthomasEMP** [reply]
the borring solution was to remove mutiny from GraphQL-Resource...as stated here: https://github.com/t1/smallrye-graphql-federation-demo

**Comment by kushchotai786** [comment]
Hello,
Just following up, Do we have any update on out-of-the-box support for Quarkus?

**Reply by phillip-kruger** [reply]
Do you mean support for Mutiny ? Everything should work for non Mutiny, but supporting Mutiny still needs to be added

**Reply by kushchotai786** [reply]
yes, supporting Mutiny. Do we have any estimate?

**Reply by phillip-kruger** [reply]
No, but we accept pull requests :)
